ikasaneip/solr/solr-client/src/main/java/org/ikasan/error/reporting/service/SolrErrorReportingManagementServiceImpl.java

```diff
@@ -17,7 +18,7 @@
 /**
  * Created by Ikasan Development Team on 23/09/2017.
  */
-public class SolrErrorReportingManagementServiceImpl extends SolrServiceBase implements SolrService<ErrorOccurrence>, ErrorReportingService<ErrorOccurrence<byte[]>, ErrorOccurrence>
+public class SolrErrorReportingManagementServiceImpl extends SolrServiceBase implements SolrService<ErrorOccurrence>, ErrorReportingService<ErrorOccurrence<byte[]>, ErrorOccurrence>, BatchInsert<ErrorOccurrence>
```

Comment:
Maybe I am missing something in here.

But why do we need 2 different Service Types for ErrorReportingManagementServiceImpl.

Should the service be same and only we swap out DAO implmentation once for Solr one for hibernate ?

Reply:
I think batch insert gives as a higher level of abstraction that can be separate from a full DAO implementation purely for data consumption.